db/migrate/20191123143443_change_user_to_ons_constituency.rb

```diff
@@ -0,0 +1,19 @@
+require_relative "../fixtures/ons_constituency_lookup"
+
+class ChangeUserToOnsConstituency < ActiveRecord::Migration[5.2]
+  def up
+    cons_id_to_ons_lookup = OnsConstituencyLookup.new
+
+    User.all.each do |user|
+      unless user.constituency_id.nil?
+        ons_id = cons_id_to_ons_lookup.find_by_id(user.constituency_id)
+        user.update!(constituency_ons_id: ons_id) unless ons_id.nil?
+      end
+    end
+    rename_column :users, :constituency_id, :old_constituency_id
```

Comment:
Worth noting that this rename is not technically necessary and only there to discourage users from using the deprecated column.

We discovered this can cause a crash:

    SQLite3::ConstraintException: FOREIGN KEY constraint failed

This is because sqlite3 is incapable of renaming columns, so Rails emulates a rename by:

- creating a new users table with the column rename,
- copying data to it,
- dropping the old one, and then
- renaming the table.

However the attempt to drop the old `users` table will fail with the above error if there are any rows in the `mobile_phones` table with a `user_id` foreign key referencing rows in `users`.

This is a known issue e.g. https://stackoverflow.com/questions/25392012/change-column-name-in-sqlite-when-other-tables-have-foreign-keys-to-it which shows that one workaround is to run [`PRAGMA foreign_keys = off`](https://www.sqlite.org/pragma.html#pragma_foreign_keys) before running the migration.  However this only happens in dev so we can note the workaround and move on.